src/verilog/verilog_typecheck.cpp

```diff
@@ -203,18 +203,19 @@ Function: verilog_typecheckt::typecheck_builtin_port_connections
 void verilog_typecheckt::typecheck_builtin_port_connections(
   verilog_inst_baset::instancet &inst)
 {
-  exprt &range=static_cast<exprt &>(inst.add(ID_range));
+  exprt &range_expr = static_cast<exprt &>(inst.add(ID_range));
 
-  mp_integer msb, lsb;
-  
-  if(range.is_nil() || range.id()=="")
-    msb=lsb=0;
+  ranget range;
+
+  if(range_expr.is_nil() || range_expr.id() == "")
+    range.msb = range.lsb = 0;
   else
-    convert_range(range, msb, lsb);
+    range = convert_range(range_expr);
+
+  if(range.lsb > range.msb)
+    std::swap(range.lsb, range.msb);
```

Comment:
Why is this here (and just here) the right thing to do?

Reply:
It's probably broken. I'll investigate.